### PR TITLE
Adding a Convolutional Autoencoder and a Variational Autoencoder (VAE)

### PR DESCRIPTION
**Pull request by Luozeyuisthebest**
I added a Convolutional Autoencoder and a Variational Autoencoder (VAE). I followed the previous code structure and only added code on this basis. I have trained these two new models and taken screenshots of their results. After modifying these files, I ran the previous two models (Sparse Autoencoder and Autoencoder) again and recorded their performances. Finally, I revised the readme document and added the code required for the new model to run. The following are the changes I have made
I added two new training functions to the file. model.py
2. I added two loss functions that the new model will use in the file. train.py
3. I added the code needed to run the new model in the file. readme.md
4. I added the necessary comments for all the new code
5.My updated code didn't break existing structure. No monoliths were made and dataset was being called neatly from file.  dataset.py

Explanation of the performance of the new model:
The performance of the Convolutional Autoencoder is better than that of previous models. The loss at the 20th Epoch is around 0.004, and it can regenerate almost perfect black-and-white images.
The performance of Variational Autoencoder is somewhat poor. In fact, when the standard VAE model was initially used, a Mode Collapse occurred, and all the generated images were the same. Therefore, I increased the value of beta to 8, used a deeper network, switched to BCE loss, and enhanced the KL constraint. Now the model can reconstruct the image to a certain extent (although some numbers are still confused).

### CODE REVIEW

**Comment by Luozeyuisthebest** [comment]
conv_autoencoder
conv
conv_loss

**Comment by Luozeyuisthebest** [comment]
VAE
VAE
VAE_loss

**Comment by Luozeyuisthebest** [comment]
AutoEncoder
AutoEncoder
AutoEncoder_loss

**Comment by Luozeyuisthebest** [comment]
Spare
Spare
Spare_loss

**Comment by sleepingcat4** [comment]
Why you added training function inside model.py file? 

+ Why have you added the loss function inside the train.py file? 

Because loss function should be part of the model itself + training function should be inside the train.py function if it's too much modification then should be done by creating a new function inside the train.py file @Luozeyuisthebest

**Comment by sleepingcat4** [comment]
@Luozeyuisthebest did you do the work?btw you should mention me on the replies otherwise my phone doesn't send me a notification and I miss them.

**Comment by Luozeyuisthebest** [comment]
What I mean is not to modify the existing train function, but to encapsulate all the changes within the model.
I have already made the modifications. There are quite a few differences from the version being requested. I will pull a new request

**Comment by sleepingcat4** [comment]
@Luozeyuisthebest I'm confused, I mean you can follow my pull request to know what I did, I will check your pull request after a few hours when I am back
